data_retriever/event_queue.py

```
@@ -0,0 +1,67 @@
+
+
+STATE = "migration:state"
+EVENTS = "migration:events"
+
+class EventQueue:
+    def __init__(self):
+        try:
+            self._redis = Redis()
+            self._redis.ping()
+        except Exception as e:
+            raise ConnectionError(f"Failed to connect to Redis: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)